### PR TITLE
Remove `metabase.async.api-response`

### PR DESCRIPTION
**Pull request by camsaul**
When working on #27096 I started to wonder if we really needed the extra layer of indirection we got from having a separate `output-chan` as opposed to just writing to our OutputStream directly. I pulled on that thread until the whole sweater came unraveled and I finally just concluded that this namespace is basically completely unused at this point in time. 

This code was originally written to handle `defendpoint`s that returned `core.async` channels -- which we don't do anymore since we switched to using Jetty async handlers directly with the newer `metabase.query-processor.streaming` namespace. I had written that `metabase.async.api-response` was deprecated in 0.35.0 in favor of `metabase.query-processor.streaming` when we shipped the new implementation... I just didn't realize that we're finally at the point where we can now remove it entirely. 

The only effect this has is removing dead code and super flaky tests

### CODE REVIEW

File: src/metabase/server/handler.clj [hunk 1]
```diff
@@ -1,26 +1,46 @@
...
+   [metabase.server.middleware.offset-paging :as mw.offset-paging]
+   [metabase.server.middleware.security :as mw.security]
+   [metabase.server.middleware.session :as mw.session]
+   [metabase.server.middleware.ssl :as mw.ssl]
+   [metabase.server.routes :as routes]
+   [ring.core.protocols :as ring.protocols]
+   [ring.middleware.cookies :refer [wrap-cookies]]
+   [ring.middleware.gzip :refer [wrap-gzip]]
+   [ring.middleware.keyword-params :refer [wrap-keyword-params]]
+   [ring.middleware.params :refer [wrap-params]]))
```
Comment (camsaul) [inline src/metabase/server/handler.clj:20]: This is the new way to indent `:require`s. New rule is to reindent stuff this way as you make changes to stuff

File: src/metabase/server/handler.clj [hunk 2]
```diff
@@ -1,26 +1,46 @@
...
+  (write-body-to-stream [bool response output-stream]
+    (ring.protocols/write-body-to-stream (str bool) response output-stream))
+
+  clojure.lang.Keyword
+  (write-body-to-stream [kkey response output-stream]
+    (ring.protocols/write-body-to-stream
+     (if-let  [key-ns (namespace kkey)]
+       (str key-ns "/" (name kkey))
+       (name kkey))
+     response output-stream)))
```
Comment (camsaul) [inline src/metabase/server/handler.clj:43]: These were the only parts of `metabase.async.api-response` actually still needed. They have nothing to do with anything being async

File: test/metabase/test/util/async.clj [hunk 3]
```diff
@@ -1,23 +1,6 @@
 (ns metabase.test.util.async
-  (:require [clojure.core.async :as a])
-  (:import java.util.concurrent.TimeoutException))
-
-(defn wait-for-close
```
Comment (camsaul) [inline test/metabase/test/util/async.clj:5]: This was only used in the tests for this namespace and `wait-for-result` in the same namespace accomplished the same thing

File: dev/src/user.clj [hunk 4]
```diff
@@ -1,4 +1,9 @@
-(ns user)
+(ns user
+  (:require
+   [metabase.test-runner.assert-exprs]))
+
+;; make sure stuff like `schema=` and what not are loaded
+(comment metabase.test-runner.assert-exprs/keep-me)
```
Comment (camsaul) [inline dev/src/user.clj:6]: Not directly related but I did get into a situation where I started a REPL and started running tests in `metabase.async.api-response-test` and ran into "no method for `schema=`" or whatever errors so adding this seems like a good idea

File: src/metabase/async/util.clj [hunk 5]
```diff
@@ -35,7 +37,7 @@
       (a/close! out-chan)))
   nil)
 
-(defn cancelable-thread-call
+(defn- cancelable-thread-call
```
Comment (camsaul) [inline src/metabase/async/util.clj:40]: Only used in this namespace so I made it private. I think a lot of this stuff in this namespace is pretty close to being able to be removed

**Review by metamben — APPROVED** [review]
🎉

**Pull request merged**